src/smolagents/models.py

```diff
@@ -1161,6 +1170,136 @@ def create_client(self):
         return openai.AzureOpenAI(**self.client_kwargs)
 
 
+class AmazonBedrockServerModel(ApiModel):
+    """
+    A model class for interacting with Amazon Bedrock Server models through the Bedrock API.
+
+    This class provides an interface to interact with various Bedrock language models,
+    allowing for customized model inference, guardrail configuration, message handling,
+    and other parameters allowed by boto3 API.
+
+    Parameters:
+        model_id (`str`):
+            The model identifier to use on Bedrock (e.g. "us.amazon.nova-pro-v1:0").
+        boto3_client (`boto3.client`, *optional*):
+            A custom boto3 client for AWS interactions. If not provided, a default client will be created.
+        inference_config (`dict`, *optional*):
+            Configuration parameters for model inference, such as temperature, max tokens, and other model-specific settings.
+        guardrail_config (`dict`, *optional*):
+            Configuration for model guardrails to control and moderate model behavior and outputs.
+        api_kwargs (`dict`, *optional*):
+            Additional keyword arguments to be passed to boto3 bedrock-runtime API requests.
```

Comment:
Note that there is no need to explicitly enumerate all model-specific inference kwargs: all additional kwargs passed to model instantiation will be passed to model inference. 
- You can add `**kwargs` to the docstring and explain this

Reply:
Yeah, make sense! I changed this + added more examples in the docstring.